packages/swayze/lib/src/core/controller/selection/selection_controller.dart

```diff
@@ -2,12 +2,15 @@ import 'package:flutter/foundation.dart';
 import 'package:flutter/rendering.dart';
 
 import '../controller.dart';
+import 'user_selections/fill_selection_state.dart';
 
 export 'model/selection.dart';
 export 'model/selection_style.dart';
 export 'user_selections/model.dart';
 export 'user_selections/user_selection_state.dart';
 
+typedef _StateChanger<T> = T Function(T previousState);
```

Comment:
well we are changing the controller state, but I don't think the `Changer` noun fits well here, it seems odd. What about `Transformer`?

```suggestion
typedef _StateTransformer<T> = T Function(T previousState);
```

Reply:
Why? We change states. Maybe adding a Callback suffix? What do you think? It's more descriptive.

Reply:
Changed this to `SwayzeSelectionChangerCallback` to make it also visible in other files (in case you want to use it to define a callback separately and then assign it).